internal/manifests/internal/gateway/gateway-tenants.yaml

```diff
@@ -1,9 +1,100 @@
 tenants:
-- name: telemeter
-  id: FB870BF3-9F3A-44FF-9BF7-D7A047A52F43
-  oidc:
-    clientID: telemeter
-    clientSecret: test
-    issuerURL: http://127.0.0.1:5556/dex
-    redirectURL: http://localhost:8080/oidc/telemeter/callback
-    usernameClaim: email
\ No newline at end of file
+{{- if $l := . -}}
+{{- if eq $l.Stack.Tenants.Mode "static" -}}
+{{- range $spec := $l.Stack.Tenants.Authentication }}
+- name: {{ $spec.Name }}
+  id: {{ $spec.ID }}
+  oidc:
+    {{- range $secret := $l.GatewaySecret }}
+    {{- if eq $secret.TenantName $spec.Name -}}
+    {{ if $secret.ClientID }}
+    clientID: {{ $secret.ClientID }}
+    {{- end -}}
+    {{ if $secret.ClientSecret }}
+    clientSecret: {{ $secret.ClientSecret }}
+    {{- end -}}
+    {{ if $secret.IssuerCAPath }}
+    issuerCAPath: {{ $secret.IssuerCAPath }}
+    {{- end -}}
+    {{- end -}}
+    {{- end -}}
+    {{ print "\n" }}
+    issuerURL: {{ $spec.OIDC.IssuerURL }}
+    redirectURL: {{ $spec.OIDC.RedirectURL }}
+    {{ if $spec.OIDC.UsernameClaim }}
+    usernameClaim: {{ $spec.OIDC.UsernameClaim }}
+    {{- end -}}
+    {{ if $spec.OIDC.GroupClaim }}
+    groupClaim: {{ $spec.OIDC.GroupClaim }}
+    {{- end -}}
+{{- end -}}
+{{- else if eq $l.Stack.Tenants.Mode "dynamic" -}}
+{{- if $tenant := $l.Stack.Tenants -}}
+{{- range $spec := $tenant.Authentication }}
+- name: {{ $spec.Name }}
+  id: {{ $spec.ID }}
+  oidc:
+    {{- range $secret := $l.GatewaySecret }}
+    {{- if eq $secret.TenantName $spec.Name -}}
+    {{ if $secret.ClientID }}
+    clientID: {{ $secret.ClientID }}
+    {{- end -}}
+    {{ if $secret.ClientID }}
+    clientSecret: {{ $secret.ClientSecret }}
+    {{- end -}}
+    {{ if $secret.ClientID }}
+    issuerCAPath: {{ $secret.IssuerCAPath }}
+    {{- end -}}
+    {{- end -}}
+    {{- end -}}
+    {{ print "\n" }}
+    issuerURL: {{ $spec.OIDC.IssuerURL }}
+    redirectURL: {{ $spec.OIDC.RedirectURL }}
+    {{ if $spec.OIDC.UsernameClaim }}
+    usernameClaim: {{ $spec.OIDC.UsernameClaim }}
+    {{- end -}}
+    {{ if $spec.OIDC.GroupClaim }}
+    groupClaim: {{ $spec.OIDC.GroupClaim }}
+    {{- end -}}
+  {{ print "\n" }}
+  opa:
+    url: {{ $tenant.Authorization.OPA.URL }}
+{{- end -}}
+{{- end -}}
+{{- else -}}
+{{ print "\n" }}
+- name: application
+  id: 32e45e3e-b760-43a2-a7e1-02c5631e56e9
+  oidc:
+    clientID: test
+    clientSecret: ZXhhbXBsZS1hcHAtc2VjcmV0
```

Comment:
Please add a TODO per tenant here that these need to be replaced once we integrate dex into lokistack-gateway.